kijiji_repost_headless/kijiji_api.py

```diff
@@ -207,11 +207,24 @@ def post_ad_using_data(self, data, image_files=[]):
 
     def get_all_ads(self):
         """
-        Return an iterator of tuples containing the ad title and ad ID for every ad
+        Return a list of dicts with properties for every active ad
         """
         resp = self.session.get('https://www.kijiji.ca/my/ads.json')
         resp.raise_for_status()
         ads_json = json.loads(resp.text)
-        ad_ids = [entry['id'] for entry in ads_json['ads'].values()]
-        ad_names = [entry['title'] for entry in ads_json['ads'].values()]
-        return zip(ad_names, ad_ids)
+        ads_info = ads_json['ads']
+
+        # Get rank (ie. page number) for each ad
+        # Construct url in the form of "https://www.kijiji.ca/my/ranks?ids=<ad_id1>&ids=<ad_id2>&..."
+        # where each ad ID given returns its page rank
+        query_str = "&".join("ids={}".format(ad_id) for ad_id in [ad['id'] for ad in ads_info.values()])
```

Comment:
Can we use a library? Like urllib? Or maybe requests supports this natively already

Reply:
I had thought of this, but for some reason I was thinking that I couldn't do it with any of the common libraries. Turns out that requests has support for [passing in query params](http://docs.python-requests.org/en/master/user/quickstart/#passing-parameters-in-urls) 😆 

I'll push a new commit shortly.